File: src/main/java/org/luxons/sevenwonders/game/cards/HandRotationDirection.java
package org.luxons.sevenwonders.game.cards;

public enum HandRotationDirection {
    LEFT(-1), RIGHT(1);

    private final int indexOffset;

    HandRotationDirection(int i) {
        this.indexOffset = i;
    }

    public int getIndexOffset() {
        return indexOffset;
    }
}

File: game-engine/src/main/kotlin/org/luxons/sevenwonders/game/cards/HandRotationDirection.kt
package org.luxons.sevenwonders.game.cards

enum class HandRotationDirection {
    LEFT,
    RIGHT;

    companion object {
        // clockwise (pass to the left) at age 1, and alternating
        fun forAge(age: Int): HandRotationDirection = if (age % 2 == 0) RIGHT else LEFT
    }
}
